iina/PlayerCore.swift

```diff
@@ -579,17 +580,19 @@ class PlayerCore: NSObject {
     info.disableOSDForFileLoading = true
     guard let path = mpvController.getString(MPVProperty.path) else { return }
     info.currentURL = URL(fileURLWithPath: path)
-    // add files in same folder
-    if ud.bool(forKey: Preference.Key.playlistAutoAdd) {
-      autoLoadFilesInCurrentFolder()
-    }
-    // auto load matched subtitles
-    if let matchedSubs = info.matchedSubs[path] {
-      for sub in matchedSubs {
-        loadExternalSubFile(sub)
+    backgroundQueue.async {
```

Comment:
It only takes considerable time when calculating edit distance between filenames. The complexity is like `O(mn)`, so e.g. if you have 1000 videos and 1 sub, it will calculate 1000 times, but if there are 1000 videos and subs respectively, it will run 1,000,000 times (which takes 15s on my mbp). I'll check and see if I can reproduce this problem.